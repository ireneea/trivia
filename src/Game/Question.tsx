import React from "react";
import { View, Text, StyleSheet, AccessibilityProps } from "react-native";

import { globalStyles, fontSize, colors } from "../styles/globals";
import { QuestionType } from "../ts/appTypes";

type Props = {
  question?: QuestionType;
};

const Question: React.FC<Props & AccessibilityProps> = ({ question }) => {
  const text = question?.text;

  return (
    <View testID="question" style={styles.questionContainer}>
      <Text style={styles.questionText} accessibilityLabel={text}>
        {text}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  questionContainer: StyleSheet.flatten([globalStyles.centredContainer, { alignItems: "flex-start" }]),
  questionText: {
    paddingHorizontal: 15,
    textAlign: "center",
    color: colors.normal,
    width: "100%",
    ...fontSize.xxxl,
  },
});

export default Question;
