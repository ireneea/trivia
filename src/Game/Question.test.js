import React from "react";
import { render } from "@testing-library/react-native";

import Question from "./Question";

describe("Question", () => {
  it("render Question correctly", () => {
    const { baseElement } = render(<Question />);
    expect(baseElement).toMatchSnapshot();
  });

  it("displays the question", () => {
    const question = { text: "one" };
    const { getByText } = render(<Question question={question} />);
    expect(getByText(question.text)).toBeTruthy();
  });
});
