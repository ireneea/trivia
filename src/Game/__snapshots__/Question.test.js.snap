// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question render Question correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="question"
    >
      <Text />
    </View>
  </View>
</View>
`;
